backend/ml/handle_data.py

```diff
@@ -23,32 +23,35 @@
 """
 
 
-def select_criteria(comparison_data, crit):
+def select_criteria(comparison_data: Iterable[Tuple[int, int, int, str, float, float]],
+                    criteria_name: str):
     """Extracts not None comparisons of one criteria
 
-    comparison_data: output of fetch_data()
-    crit: str, name of criteria
+    comparison_data: output of fetch_data(), can be an iterable of tuple of size 6
+    criteria_name: str
 
     Returns:
     - list of all ratings for this criteria
         ie list of [contributor_id: int, video_id_1: int, video_id_2: int,
                     criteria: str (crit), score: float, weight: float]
     """
+    # TODO Check if l_ratings is used as a list, not as an iterable or sequence.
     l_ratings = [
-        comp for comp in comparison_data if (comp[3] == crit and comp[4] is not None)
+        comparison for comparison in comparison_data
+        if comparison[3] == criteria_name and comparison[4] is not None
     ]
     return l_ratings
 
 
-def shape_data(l_ratings):
+def shape_data(l_ratings: Iterable[Tuple[int, int, int, str, int, int]]):
     """Shapes data for distribute_data()/distribute_data_from_save()
 
     l_ratings : list of not None ratings ([0,100]) for one criteria, all users
 
     Returns : one array with 4 columns : userID, vID1, vID2, rating ([-1,1])
     """
-    l_clear = [rating[:3] + [rescale_rating(rating[4])] for rating in l_ratings]
-    return np.asarray(l_clear)
```

Comment:
https://stackoverflow.com/questions/14415741/what-is-the-difference-between-numpys-array-and-asarray-functions#comment36996511_14415801
Here, l_clear is a list, so a copy must be made.

If my work isn't conclusive, we still should replace this to a np.array call